src/main/scala/pipedsl/common/Utilities.scala

```diff
@@ -305,6 +307,167 @@ object Utilities {
     }
   }
 
+
+  def opt_func[A, B](f :A => B) : Option[A] => Option[B] =
+    {
+      case Some(value) => Some(f(value))
+      case None => None
+    }
+
+  /**
+   * Maps the function f_opt over the types of e1.
+   * MODIFIES THE TYPE OF e1
+   * @param e1 the expression to map over
+   * @param f_opt the function to apply to the types
+   * @return the expression with new types
+   */
+  private def typeMapExpr(e1 :Expr, f_opt : Option[Type] => Option[Type]) : Expr =
+    {
+      try
+        {
+          e1.typ = f_opt(e1.typ)
+        } catch
+        {
+          case _ :scala.MatchError =>
+            e1 match {
+              case EInt(v, _, _) =>
+//                println(s"DEFAULTING ON INT. CURRENTLY: ${e1.typ}")
+                val sign :TSignedNess = if(e1.typ.isDefined && e1.typ.get.isInstanceOf[TSizedInt])
```

Comment:
Same comment as in the BSV file -> would be better to have a matching function that errors rather than explicit runtime casts.